### PR TITLE
General: Split wal into multiple files.

### PR DESCRIPTION
**Pull request by dylanaraps**
This PR extends #7 and splits `wal` into multiple files based on their purpose in the original script. This will make `wal` much easier to maintain in the future.

@danielx, can you take a look at this?

### CODE REVIEW

**Comment by danielx** [comment]
Looks good, but i would make two changes:

* Remove the use of globals to make the functions easier to test and reason about.
* Make module version available in`__init__.py` by import or declaring it there.

**Comment by dylanaraps** [comment]
> Remove the use of globals to make the functions easier to test and reason about.

I'm currently in the process of doing this.

> Make module version available in__init__.py by import or declaring it there.

Good idea, I'll get onto this.

Thanks for the advice!

**Comment by dylanaraps** [comment]
I've removed all globals (minus the 3 constants which are ok imo.) and I've added `__version__` to `__init__.py`. 

How does this PR look now? @danielx

File: pywal/gen_colors.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        util.save_file("\n".join(colors), cache_file)
+
+        print("colors: Generated colorscheme")
+        if not quiet:
+            util.disown("notify-send", "wal: Generation complete.")
+
+    return colors
+
+
+def sort_colors(colors):
```
Comment (danielx) [inline pywal/gen_colors.py]: Function body could be shortened to

```python
return colors[:1] + colors[9:] + colors[8:]
```

And If the length of `colors` might be greater than 16
```python
return colors[:1] + colors[9:16] + colors[8:16]
```

**Comment by danielx** [comment]
Looks good to me 👍

**Comment by dylanaraps** [comment]
Thanks for taking another look I apreciate it!

**Pull request merged**